### PR TITLE
Make get_wavelength error out if data has no shape

### PR DESCRIPTION
**Pull request by emolter**
Relates to [JP-4228](https://jira.stsci.edu/browse/JP-4228)

This PR enforces that `lib.wcs_utils.get_wavelength` raises an error if the data array is empty, and updates one unit test that passed empty data into that function.

https://github.com/spacetelescope/RegressionTests/actions/runs/21496516296

## Tasks
- [ ] If you have a specific reviewer in mind, tag them.
- [x] add a build milestone, i.e. `Build 12.0` (use the [latest build](https://github.com/spacetelescope/jwst/milestones) if not sure)
- [x] Does this PR change user-facing code / API? (if not, label with `no-changelog-entry-needed`)
  - [ ] write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see [changelog readme](https://github.com/spacetelescope/jwst/blob/main/changes/README.rst) for instructions) 
    - if your change breaks **step-level or public API** ([as defined in the docs](https://jwst.readthedocs.io/en/latest/jwst/user_documentation/more_information.html#api-public-vs-private)), also add a `changes/<PR#>.breaking.rst` news fragment
  - [x] update or add relevant tests
  - [ ] update relevant docstrings and / or `docs/` page
  - [x] [start a regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/jwst.yml) and include a link to the running job ([click here for instructions](https://github.com/spacetelescope/RegressionTests/blob/main/docs/running_regression_tests.md))
    - [ ] Do truth files need to be updated ("okified")?
      - [ ] **after the reviewer has approved these changes**, run `okify_regtests` to update the truth files
- [ ] if a JIRA ticket exists, [make sure it is resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Review by melanieclarke — APPROVED** [review]
Sounds sensible to me!

File: jwst/lib/wcs_utils.py [hunk 1]
```diff
@@ -33,6 +33,8 @@ def get_wavelengths(model, exp_type="", order=None, use_wavecorr=None):
     wl_array : 2-D ndarray
         An array of wavelengths corresponding to the data in ``model``.
     """
+    if model.data.size == 0:
```
Comment (pllim) [inline jwst/lib/wcs_utils.py]: Is this enough? A scalar ndarray will still pass this check but have shape of `()` which then will crash later when the shape is used.

Comment (emolter) [reply jwst/lib/wcs_utils.py]: looking at this function again I don't think it does anything sensible unless `data` has at least 2 dimensions.  I updated the check in https://github.com/spacetelescope/jwst/pull/10205/commits/f9055d71b74e31cd4ed7539ffe0d16564dc2c4bf

File: jwst/lib/tests/test_wcs_utils.py [hunk 2]
```diff
@@ -80,6 +81,15 @@ def test_get_wavelengths():
     assert_allclose(wl, wl_og)
 
 
+def test_get_wavelengths_empty():
+    # create a mock SlitModel
+    model = create_model()
+    model.data = np.array([])
```
Comment (pllim) [inline jwst/lib/tests/test_wcs_utils.py]: Should we also add checks for `np.array(1)` (scalar) and a 1D array, just in case?

File: jwst/lib/tests/test_wcs_utils.py [hunk 3]
```diff
@@ -80,6 +81,15 @@ def test_get_wavelengths():
...
+
+    with pytest.raises(ValueError):
```
Comment (pllim) [inline jwst/lib/tests/test_wcs_utils.py]:
```suggestion
    with pytest.raises(ValueError, match=".*cannot compute wavelengths"):
```

Comment (emolter) [reply jwst/lib/tests/test_wcs_utils.py]: https://github.com/spacetelescope/jwst/pull/10205/commits/4b06c2416513e5a50d36724af96a4192417a0031

Not sure any of these cases could be hit in a step/pipeline run in practice because the datamodel types passed in here typically validate the data against a shape or an ndim schema specification.  But I guess because the documented input type is `JwstDataModel` there's some silly edge case where this could fail if used outside pipeline

File: jwst/lib/tests/test_wcs_utils.py [hunk 4]
```diff
@@ -80,6 +81,22 @@ def test_get_wavelengths():
...
+
+    model.data = np.array([])
+    with pytest.raises(ValueError, match=".*cannot compute wavelengths"):
+        get_wavelengths(model)
+
+    model.data = np.array(1)
+    with pytest.raises(ValueError, match=".*cannot compute wavelengths"):
+        get_wavelengths(model)
+
+    model.data = np.ones((10))
```
Comment (pllim) [inline jwst/lib/tests/test_wcs_utils.py]: Sorry, I should have been clearer. I think this can be simplified via `pytest.mark.parametrize`.

**Comment by pllim** [comment]
Re: https://github.com/spacetelescope/jwst/pull/10205#discussion_r2747400863 -- Right. But since now you have to add a check, might as well catch the extra cases?

File: jwst/lib/tests/test_wcs_utils.py (hunk 4 above)
Comment (emolter) [reply jwst/lib/tests/test_wcs_utils.py]: I should have slowed down. using parametrize is indeed very sensible here

Comment (emolter) [reply jwst/lib/tests/test_wcs_utils.py]: https://github.com/spacetelescope/jwst/pull/10205/commits/fcd795d8e3dd19102564741538b57bffd53eda28

**Review by pllim — APPROVED** [review]
LGTM. Thanks!
